### PR TITLE
Add resize (lowres) capability to cuvid

### PR DESCRIPTION
**Pull request by arrmo**
Works great with my test file, slight errors if too aggressive with lowres on ten-copy.mkv ... but that's expected if you push it too far, agreed?

### CODE REVIEW

**Comment by erikkaashoek** [comment]
I need to check the logic around is_h264, 
do not merge yet

File: mpeg2dec.c [hunk 1]
```diff
@@ -1685,25 +1685,19 @@ int stream_component_open(VideoState *is, int stream_index)
 #endif
         }
 
-        if (codecCtx->codec_id == AV_CODEC_ID_H264) {
```
Comment (erikkaashoek) [inline mpeg2dec.c]: No need to change this

File: mpeg2dec.c [hunk 2]
```diff
@@ -1685,25 +1685,19 @@ int stream_component_open(VideoState *is, int stream_index)
 #endif
         }
 
-        if (codecCtx->codec_id == AV_CODEC_ID_H264) {
+        if ((!use_cuvid) && (codecCtx->codec_id == AV_CODEC_ID_H264)) {
+        	// SW decoder for h264, cannot resize to lower resolution
             is_h264 = 1;
 #ifdef DONATOR
 #else
-            Debug(0, "h.264 video can only be processed at full speed by the Donator version\n");
```
Comment (erikkaashoek) [inline mpeg2dec.c]: No need to change this

File: mpeg2dec.c [hunk 3]
```diff
@@ -1635,7 +1635,7 @@ int stream_component_open(VideoState *is, int stream_index)
     AVCodecContext *codecCtx;
     AVCodec *codec;
     AVCodec *codec_hw = NULL;
-
+	char strResize[11];
```
Comment (erikkaashoek) [inline mpeg2dec.c]: Can you increase the string size for safety?

File: mpeg2dec.c [hunk 4]
```diff
@@ -1685,25 +1685,19 @@ int stream_component_open(VideoState *is, int stream_index)
...
-            int w;
-            if (lowres == 10) {
-                w = codecCtx->width;
-                lowres = 0;
-                while (w > 600) {
-                    w = w >> 1;
-                    lowres++;
-                }
-            }
+            if ((lowres == 10) && (stream_index == 0))
```
Comment (erikkaashoek) [inline mpeg2dec.c]: OK

File: mpeg2dec.c [hunk 5]
```diff
@@ -1752,7 +1746,10 @@ int stream_component_open(VideoState *is, int stream_index)
     }
 
     if (!hardware_decode) av_dict_set_int(&myoptions, "gray", 1, 0);
-
+    if ((use_cuvid) && (stream_index == 0)) {
```
Comment (erikkaashoek) [inline mpeg2dec.c]: OK

File: mpeg2dec.c (hunk 3 above)
Comment (arrmo) [reply mpeg2dec.c]: Sure. Was thinking #####x#####, so 5 digits for each. But I can update, no issue. Suggestions?

File: mpeg2dec.c (hunk 1 above)
Comment (arrmo) [reply mpeg2dec.c]: It's tied to the line below - but I had to add the use_cuvid check ... or cuvid is blocked if H.264 => so OK?

File: mpeg2dec.c (hunk 2 above)
Comment (arrmo) [reply mpeg2dec.c]: That's fine, I can undo it. It was just to clarify, confused by me initially (as it's not speed that is full, rather resolution). But I can revert, NP.

**Comment by arrmo** [comment]
Updated to your latest commit - OK to merge now?

Thanks!

**Pull request merged**